bee-message/src/output/mod.rs

```diff
@@ -199,6 +202,47 @@ impl Output {
             Self::Nft(output) => Some(output.immutable_feature_blocks()),
         }
     }
+
+    /// Checks if a sufficient storage deposit was made for the given [`Output`].
+    pub fn check_sufficient_storage_deposit(&self, config: &ByteCostConfig) -> Result<(), Error> {
+        let maybe_storage_deposit_condition = match self {
```

Comment:
If you rebase to include my recent PR, you can call `unlock_conditions` directly on the output and then map `storage_deposit_return` :)

Reply:
I need to check for the type of `Output` to return early for types that don't require storage deposits, or can I assume that `check_sufficient_storage_deposit` will only be called in a context where this is not important?